File: Python/Evaluate Reverse Polish Notation.py
class Solution:
    # @param tokens, a list of string
    # @return an integer
    def evalRPN(self, tokens):
		stack = []
		for i in range(0,len(tokens)):
			if (tokens[i][0] == "-" and len(tokens[i])>1) or (tokens[i][0] >= "0" and tokens[i][0]<="9"):
				stack.append(int(tokens[i]))
				continue
			operator1 = stack.pop()
			operator2 = stack.pop()
			if tokens[i]=="+":
				stack.append(operator2+operator1)
			elif tokens[i]=="-":
				stack.append(operator2-operator1)
			elif tokens[i]=="*":
				stack.append(operator2*operator1)
			elif tokens[i]=="/":
				stack.append(int(operator2*1.0/operator1))
		return stack.pop()
s = Solution()
s.evalRPN(["10","6","9","3","+","-11","*","/","*","17","+","5","+"])

File: Java/Evaluate_Reverse_Polish_Notation.java
class Evaluate_Reverse_Polish_Notation{
	public int evalRPN(String[] tokens) {
        if(tokens == null || tokens.length == 0)
        	return 0;
        LinkedList<String> stack = new LinkedList<String>();
        for(String s:tokens){
        	if(s.equals("+")){
        		int op1 = Integer.valueOf(stack.pop());
        		int op2 = Integer.valueOf(stack.pop());
        		stack.push((op1+op2)+"");
        	}
        	else if(s.equals("-")){
        		int op1 = Integer.valueOf(stack.pop());
        		int op2 = Integer.valueOf(stack.pop());
        		stack.push((op2-op1)+"");
        	}
        	else if(s.equals("*")){
        		int op1 = Integer.valueOf(stack.pop());
        		int op2 = Integer.valueOf(stack.pop());
        		stack.push((op1*op2)+"");
        	}
        	else if(s.equals("/")){
        		int op1 = Integer.valueOf(stack.pop());
        		int op2 = Integer.valueOf(stack.pop());
        		stack.push((op2/op1)+"");
        	}
        	else{
        		stack.push(s);
        	}
        }
        return Integer.valueOf(stack.pop());
    }
}